bots/s3_block_public_access.py

```diff
@@ -0,0 +1,122 @@
+"""
+## s3_block_public_access
+What it does: turn on S3 Bucket Block public access
+Usage: AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=<true|false>  BlockPublicPolicy_NewPUTRequests=<true|false>
+Limitations: none
+Notes:
+    -  before running this bot, ensure that your applications will work correctly without public access
+    - works with JSON - Full / Basic entity findings
+Examples:
+
+    Block public access to buckets and objects granted through new public ACLs and Bucket Policies:
+    AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=true  BlockPublicPolicy_NewPUTRequests=true
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies
+    ( granted through new public ACLs and Bucket Policies Configuration Stays the same ):
+    AUTO: s3_block_public_access
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies But not for
+    new public ACLs and Bucket Policies:
+    AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=false  BlockPublicPolicy_NewPUTRequests=false
+
+"""
+
+import boto3
+from botocore.exceptions import ClientError
+
+usage = 'AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=<true|false>  BlockPublicPolicy_NewPUTRequests=<true|false> \n '
+
+text_output = ''
+
+TEXT_OUTPUT_DIC = {
+    'BlockPublicAcls': ' set to true - certain PUT requests includes a public ACL will fail \n ',
+    'IgnorePublicAcls': ' set to true -  S3 will ignore all public ACLs on this bucket and objects in this bucket \n ',
+    'BlockPublicPolicy': 'set to true -  S3 will reject calls to PUT Bucket policy if the specified bucket policy '
+                         'allow public access. \n ',
+    'RestrictPublicBuckets': 'set to true -  access to this bucket restricted to only AWS services and authorized users '
+                             'within this account if the bucket has a public policy. \n '
+}
+
+
+def run_action(boto_session, rule, entity, params):
+    # Create an S3 client
+    s3 = boto3.client('s3')
+    bucket_id = entity['id']
+
+    # For basic findings use  - entity region is not specified
```

Comment:
basic finding = basic entity ? redundant, use full

Reply:
checked with sakal , it is  redundant , removed